service/podcastService.go

```diff
@@ -715,6 +715,11 @@ func makeQuery(url string) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+	
+	setting := db.GetOrCreateSetting()
+	if len(setting.UserAgent) > 0 {
+		req.Header.Add("User-Agent", setting.UserAgent)
+	}
```

Comment:
You can just use the existing `getRequest` method from `fileService.go`. Replace the call to `NewRequest` with `getRequest` as in:

~~~
req, err := getRequest(url)
~~~